### ISSUE TITLE
[BUG] from pyarrow.feather import write_feather, FeatherReader ImportError: cannot import name 'FeatherReader' from 'pyarrow.feather'

### ISSUE THREAD

**Issue by lc822** [aertslab/pySCENIC#167]
**Describe the bug**
I have installed pyscenic using conda, however I am unable to access the help or run any commands because I get an error.

**Steps to reproduce the behavior**
1. Command run when the error occurred:

```python
pyscenic -h
```

2. Error encountered:

```pytb
/data/user/lucy/py36-v1/conda-install/envs/pyscenic/lib/python3.8/site-packages/dask/config.py:161: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
/data/user/lucy/py36-v1/conda-install/envs/pyscenic/lib/python3.8/site-packages/dask/dataframe/utils.py:13: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm
Traceback (most recent call last):
  File "/data/user/lucy/py36-v1/conda-install/envs/pyscenic/bin/pyscenic", line 5, in <module>
    from pyscenic.cli.pyscenic import main
  File "/data/user/lucy/py36-v1/conda-install/envs/pyscenic/lib/python3.8/site-packages/pyscenic/cli/pyscenic.py", line 20, in <module>
    from pyscenic.rnkdb import opendb, RankingDatabase
  File "/data/user/lucy/py36-v1/conda-install/envs/pyscenic/lib/python3.8/site-packages/pyscenic/rnkdb.py", line 12, in <module>
    from pyarrow.feather import write_feather, FeatherReader
ImportError: cannot import name 'FeatherReader' from 'pyarrow.feather' (/data/user/lucy/py36-v1/conda-install/envs/pyscenic/lib/python3.8/site-packages/pyarrow/feather.py)
```

**Expected behavior**
View the help for pyscenic.

**Please complete the following information:**
- pySCENIC version: 0.10.0
- Installation method: pip
- Run environment: CLI
- OS: Red Hat 4.4.7-23
- Package versions: 

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       0_gnu    conda-forge
abseil-cpp                20200225.2           he1b5a44_0    conda-forge
arboreto                  0.1.5                    pypi_0    pypi
arrow-cpp                 0.17.0           py38h9d6855f_0    conda-forge
attrs                     19.3.0                     py_0    conda-forge
aws-sdk-cpp               1.7.164              hc831370_1    conda-forge
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.5              pyh9f0ad1d_0    conda-forge
boltons                   20.1.0                   pypi_0    pypi
boost-cpp                 1.72.0               h8e57a91_0    conda-forge
brotli                    1.0.7             he1b5a44_1001    conda-forge
brotlipy                  0.7.0           py38h1e0a361_1000    conda-forge
bzip2                     1.0.8                h516909a_2    conda-forge
c-ares                    1.15.0            h516909a_1001    conda-forge
ca-certificates           2020.4.5.1           hecc5488_0    conda-forge
certifi                   2020.4.5.1       py38h32f6830_0    conda-forge
cffi                      1.14.0           py38hd463f26_0    conda-forge
chardet                   3.0.4           py38h32f6830_1006    conda-forge
click                     7.1.2                    pypi_0    pypi
cloudpickle               1.4.1                    pypi_0    pypi
cryptography              2.9.2            py38h766eaa4_0    conda-forge
curl                      7.69.1               h33f0ec9_0    conda-forge
cytoolz                   0.10.1                   pypi_0    pypi
dask                      1.0.0                    pypi_0    pypi
decorator                 4.4.2                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
dill                      0.3.1.1                  pypi_0    pypi
distributed               1.28.1                   pypi_0    pypi
entrypoints               0.3             py38h32f6830_1001    conda-forge
frozendict                1.2                      pypi_0    pypi
gflags                    2.2.2             he1b5a44_1002    conda-forge
glog                      0.4.0                h49b9bf7_3    conda-forge
grpc-cpp                  1.28.1               h8e748ff_2    conda-forge
h5py                      2.10.0                   pypi_0    pypi
heapdict                  1.0.1                    pypi_0    pypi
icu                       64.2                 he1b5a44_1    conda-forge
idna                      2.9                        py_1    conda-forge
importlib-metadata        1.6.0            py38h32f6830_0    conda-forge
importlib_metadata        1.6.0                         0    conda-forge
interlap                  0.2.6                    pypi_0    pypi
ipykernel                 5.2.1            py38h23f93f0_0    conda-forge
ipython                   7.14.0           py38h32f6830_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.0           py38h32f6830_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
joblib                    0.15.1                   pypi_0    pypi
json5                     0.9.0                      py_0    conda-forge
jsonschema                3.2.0            py38h32f6830_1    conda-forge
jupyter_client            6.1.3                      py_0    conda-forge
jupyter_core              4.6.3            py38h32f6830_1    conda-forge
jupyterlab                2.1.2                      py_0    conda-forge
jupyterlab_server         1.1.3                      py_0    conda-forge
krb5                      1.17.1               h2fd8d38_0    conda-forge
ld_impl_linux-64          2.34                 h53a641e_0    conda-forge
libblas                   3.8.0               14_openblas    conda-forge
libcblas                  3.8.0               14_openblas    conda-forge
libcurl                   7.69.1               hf7181ac_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libevent                  2.1.10               h72c5cf5_0    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.2.0                h24d8f2e_2    conda-forge
libgfortran-ng            7.5.0                hdf63c60_6    conda-forge
libgomp                   9.2.0                h24d8f2e_2    conda-forge
liblapack                 3.8.0               14_openblas    conda-forge
libopenblas               0.3.7                h5ec1e0e_6    conda-forge
libprotobuf               3.11.4               h8b12597_0    conda-forge
libsodium                 1.0.17               h516909a_0    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.2.0                hdf63c60_2    conda-forge
llvmlite                  0.32.1                   pypi_0    pypi
loompy                    3.0.6                    pypi_0    pypi
lz4-c                     1.9.2                he1b5a44_1    conda-forge
markupsafe                1.1.1            py38h1e0a361_1    conda-forge
mistune                   0.8.4           py38h1e0a361_1001    conda-forge
msgpack                   1.0.0                    pypi_0    pypi
multiprocessing-on-dill   3.5.0a4                  pypi_0    pypi
nbconvert                 5.6.1            py38h32f6830_1    conda-forge
nbformat                  5.0.6                      py_0    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
networkx                  2.4                      pypi_0    pypi
notebook                  6.0.3            py38h32f6830_0    conda-forge
numba                     0.49.1                   pypi_0    pypi
numpy                     1.18.4           py38h8854b6b_0    conda-forge
numpy-groupies            0+unknown                pypi_0    pypi
openssl                   1.1.1g               h516909a_0    conda-forge
packaging                 20.1                       py_0    conda-forge
pandas                    1.0.3            py38hcb8c335_1    conda-forge
pandoc                    2.9.2.1                       0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.7.0              pyh9f0ad1d_0    conda-forge
pexpect                   4.8.0            py38h32f6830_1    conda-forge
pickleshare               0.7.5           py38h32f6830_1001    conda-forge
pip                       20.1               pyh9f0ad1d_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt-toolkit            3.0.5                      py_0    conda-forge
psutil                    5.7.0                    pypi_0    pypi
ptyprocess                0.6.0                   py_1001    conda-forge
pyarrow                   0.17.0           py38h56ab25f_0    conda-forge
pycparser                 2.20                       py_0    conda-forge
pygments                  2.6.1                      py_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyrsistent                0.16.0           py38h1e0a361_0    conda-forge
pyscenic                  0.10.0                   pypi_0    pypi
pysocks                   1.7.1            py38h32f6830_1    conda-forge
python                    3.8.2           he5300dc_7_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyyaml                    5.3.1                    pypi_0    pypi
pyzmq                     19.0.1           py38ha71036d_0    conda-forge
re2                       2020.05.01           he1b5a44_1    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
requests                  2.23.0             pyh8c360ce_2    conda-forge
scikit-learn              0.23.0                   pypi_0    pypi
scipy                     1.4.1                    pypi_0    pypi
send2trash                1.5.0                      py_0    conda-forge
setuptools                46.3.1           py38h32f6830_0    conda-forge
six                       1.14.0                     py_1    conda-forge
snappy                    1.1.8                he1b5a44_1    conda-forge
sortedcontainers          2.1.0                    pypi_0    pypi
sqlite                    3.30.1               hcee41ef_0    conda-forge
tbb                       2020.0.133               pypi_0    pypi
tblib                     1.6.0                    pypi_0    pypi
terminado                 0.8.3            py38h32f6830_1    conda-forge
testpath                  0.4.4                      py_0    conda-forge
threadpoolctl             2.0.0                    pypi_0    pypi
thrift-cpp                0.13.0               h62aa4f2_2    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
toolz                     0.10.0                   pypi_0    pypi
tornado                   6.0.4            py38h1e0a361_1    conda-forge
tqdm                      4.46.0                   pypi_0    pypi
traitlets                 4.3.3            py38h32f6830_1    conda-forge
umap-learn                0.4.3                    pypi_0    pypi
urllib3                   1.25.9                     py_0    conda-forge
wcwidth                   0.1.9              pyh9f0ad1d_0    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.5                h516909a_0    conda-forge
zeromq                    4.3.2                he1b5a44_2    conda-forge
zict                      2.0.0                    pypi_0    pypi
zipp                      3.1.0                      py_0    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
zstd                      1.4.4                h6597ccf_3    conda-forge
```

Is there a way for me to get around this?

Best wishes,
Lucy

**Comment by lc822** [comment]
Hi,

I managed to fix this issue by downgrading pyarrow to 0.16.0

Best,
Lucy

**Comment by cflerin** [comment]
@lc822 , glad you got it working. The newest `0.10.1` release now specifies that pyarrow<0.17.0 is required.
